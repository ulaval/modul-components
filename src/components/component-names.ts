export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
