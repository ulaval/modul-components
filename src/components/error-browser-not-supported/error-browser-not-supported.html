<m-error-template :skin="skin" :iconName="iconName" :title="title" :hints="mqAwareHints" :links="mqAwareLinks">
</m-error-template>
