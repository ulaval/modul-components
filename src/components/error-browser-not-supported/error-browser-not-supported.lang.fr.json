{
    "m-error-browser-not-supported": {
        "title": "Ce navigateur n'est pas supporté",
        "hint.primary.desktop": "Une mise à niveau de votre navigateur est nécessaire pour accéder à l'application.",
        "hint.primary.mobile": "Une mise à niveau de votre navigateur est nécessaire pour accéder à l'application. Pour ce faire, consulter la boutique de téléchargement d'application de votre appareil",
        "update-browser.desktop": "Mettre mon navigateur à niveau"
    }
}
