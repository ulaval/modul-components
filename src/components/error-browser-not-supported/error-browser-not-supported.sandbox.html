<div>
    <h5>default</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-error-browser-not-supported></m-error-browser-not-supported>
    </div>

    <br />

    <h5>With custom title</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-error-browser-not-supported :title="title"></m-error-browser-not-supported>
    </div>

    <br />

    <h5>With many custom hints</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-error-browser-not-supported :hints="manyHints"></m-error-browser-not-supported>
    </div>

    <br />

    <h5>With many custom links</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-error-browser-not-supported :links="manyLinks"></m-error-browser-not-supported>
    </div>

    <br />

    <h5>Unexpected error popup</h5>
    <m-button v-m-popup:errorpopup>pop me an error</m-button>
    <m-modal ref="errorpopup" :negativeLink="false" ok-label="Fermer">
            <m-error-browser-not-supported ></m-error-browser-not-supported>
    </m-modal>
</div>
