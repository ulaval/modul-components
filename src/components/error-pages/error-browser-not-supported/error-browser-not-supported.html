<m-message-page :state="state" :svg-name="svgName" :title="title" :hints="isMqMinS ? hintsDesktop : hintsMobile" :links="isMqMinS ? linksDesktop : linksMobile">
</m-message-page>
