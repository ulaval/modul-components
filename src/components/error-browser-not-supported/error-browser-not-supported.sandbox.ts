import Vue, { PluginObject } from 'vue';
import { Component } from 'vue-property-decorator';

import { ERROR_BROWSER_NOT_SUPPORTED_NAME } from '../component-names';
import { Link } from '../message-page/message-page';
import WithRender from './error-browser-not-supported.sandbox.html';

@WithRender
@Component
export class MErrorBrowserNotSupportedSandbox extends Vue {
    title: string = 'My custom browser not supported title';
    manyHints: string[] = ['My only custom hint', 'My second (long) custom hint.Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas urna rhoncus ipsum congue lobortis. Mauris vel neque condimentum, dignissim lectus ac, vehicula justo. Aliquam nunc leo, tristique hendrerit aliquam sagittis, scelerisque quis libero. Integer non augue nec lacus aliquet porttitor in nec lacus. Cras ultrices tellus est, condimentum gravida orci pulvinar quis. Integer eget turpis arcu. Curabitur consequat porta urna, at hendrerit justo consectetur non. Aenean venenatis ornare nulla, a vulputate erat eleifend ut.'];
    manyLinks: Link[] = [new Link('The first custom link', 'http://www.ulaval.ca', true), new Link('The second custom link', 'http://www.google.com', true)];
}

const ErrorBrowserNotSupportedSandboxPlugin: PluginObject<any> = {
    install(v, options): void {
        v.component(`${ERROR_BROWSER_NOT_SUPPORTED_NAME}-sandbox`, MErrorBrowserNotSupportedSandbox);
    }
};

export default ErrorBrowserNotSupportedSandboxPlugin;
