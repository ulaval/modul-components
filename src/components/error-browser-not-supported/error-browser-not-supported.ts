import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';

import { MediaQueries } from '../../mixins/media-queries/media-queries';
import MediaQueriesPlugin from '../../utils/media-queries/media-queries';
import { ModulVue } from '../../utils/vue/vue';
import { ERROR_BROWSER_NOT_SUPPORTED_NAME } from '../component-names';
import ErrorTemplatePlugin, { Link } from '../error-template/error-template';
import I18nPlugin from '../i18n/i18n';
import LinkPlugin from '../link/link';
import { MMessageState } from '../message/message';
import WithRender from './error-browser-not-supported.html';

@WithRender
@Component({
    mixins: [ MediaQueries ]
})
export class MErrorBrowserNotSupported extends ModulVue {

    @Prop({
        default: () => (Vue.prototype as any).$i18n.translate('m-error-browser-not-supported:title')
    })
    public title: string;

    @Prop({
        default: () => [new Link((Vue.prototype as any).$i18n.translate('m-error-browser-not-supported:update-browser.desktop'), 'http://outdatedbrowser.com/fr', true)]
    })
    public linksDesktop: Link[];

    @Prop({ default: () => [] })
    public linksMobile: Link[];

    @Prop({
        default: () => [(Vue.prototype as any).$i18n.translate('m-error-browser-not-supported:hint.primary.desktop')]
    })
    public hintsDesktop: string[];

    @Prop({
        default: () => [(Vue.prototype as any).$i18n.translate('m-error-browser-not-supported:hint.primary.mobile')]
    })
    public hintsMobile: string[];

    readonly state: string = MMessageState.Warning;

    readonly svgName: string = 'm-svg__error-browser-not-supported';
}

const ErrorBrowserNotSupported: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug(ERROR_BROWSER_NOT_SUPPORTED_NAME, 'plugin.install');
        v.use(I18nPlugin);
        v.use(LinkPlugin);
        v.use(MediaQueriesPlugin);
        v.use(ErrorTemplatePlugin);
        v.component(ERROR_BROWSER_NOT_SUPPORTED_NAME, MErrorBrowserNotSupported);
    }
};

export default ErrorBrowserNotSupported;
