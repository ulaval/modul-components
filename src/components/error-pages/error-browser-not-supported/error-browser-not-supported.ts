import Vue, { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import { MediaQueries } from '../../../mixins/media-queries/media-queries';
import { FRENCH, Messages } from '../../../utils/i18n/i18n';
import { ModulVue } from '../../../utils/vue/vue';
import { ERROR_BROWSER_NOT_SUPPORTED_NAME } from '../../component-names';
import MessagePagePlugin, { Link } from '../../message-page/message-page';
import { MMessageState } from '../../message/message';
import WithRender from './error-browser-not-supported.html';


@WithRender
@Component({
    mixins: [MediaQueries]
})
export class MErrorBrowserNotSupported extends ModulVue {

    @Prop({
        default: () => (Vue.prototype).$i18n.translate('m-error-browser-not-supported:title')
    })
    public title: string;

    @Prop({
        default: () => [new Link((Vue.prototype).$i18n.translate('m-error-browser-not-supported:update-browser.desktop'), 'http://outdatedbrowser.com/fr', true)]
    })
    public linksDesktop: Link[];

    @Prop({ default: () => [] })
    public linksMobile: Link[];

    @Prop({
        default: () => [(Vue.prototype).$i18n.translate('m-error-browser-not-supported:hint.primary.desktop')]
    })
    public hintsDesktop: string[];

    @Prop({
        default: () => [(Vue.prototype).$i18n.translate('m-error-browser-not-supported:hint.primary.mobile')]
    })
    public hintsMobile: string[];

    readonly state: string = MMessageState.Warning;

    readonly svgName: string = 'm-svg__error-browser-not-supported';
}

const ErrorBrowserNotSupported: PluginObject<any> = {
    install(v, options): void {
        v.use(MessagePagePlugin);
        v.component(ERROR_BROWSER_NOT_SUPPORTED_NAME, MErrorBrowserNotSupported);
    }
};

export default ErrorBrowserNotSupported;
