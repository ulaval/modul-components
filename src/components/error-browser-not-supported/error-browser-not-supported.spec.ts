import { shallow, Wrapper } from '@vue/test-utils';

import { renderComponent } from '../../../tests/helpers/render';
import { ERROR_TEMPLATE_NAME } from '../component-names';
import { Link } from '../error-template/error-template';
import { MErrorBrowserNotSupported } from './error-browser-not-supported';

let wrapper: Wrapper<MErrorBrowserNotSupported>;
let isMobileDevice: boolean;
const A_CUSTOM_TITLE: string = 'An error title.';
const A_HINT: string = 'aHint';
const A_HINT_MOBILE: string = 'aHint-Mobile';
const A_LINK: Link = new Link('aLabel', 'anUrl');
const A_LINK_MOBILE: Link = new Link('aLabel-Mobile', 'anUrl');

const getStubs: any = () => {
    return {
        [ERROR_TEMPLATE_NAME]: '<div><slot /></div>'
    };
};

const getMocks: any = () => {
    return {
        $mq: {
            register: jest.fn()
        }
    };
};
const initializeWrapperDefaultValues: () => void = (): void => {
    wrapper = shallow(MErrorBrowserNotSupported, {
        mocks: getMocks(),
        stubs: getStubs(),
        mixins: [{
            data: function(): any {
                return {
                    isMqMinS: isMobileDevice ? false : true
                };
            }
        }]
    });
};

const initializeWrapperCustomValues: () => void = (): void => {
    wrapper = shallow(MErrorBrowserNotSupported, {
        stubs: getStubs(),
        mocks: getMocks(),
        propsData: {
            title: A_CUSTOM_TITLE,
            hintsDesktop: [A_HINT],
            hintsMobile: [A_HINT_MOBILE],
            linksDesktop: [A_LINK],
            linksMobile: [A_LINK_MOBILE]
        },
        mixins: [{
            data: function(): any {
                return {
                    isMqMinS: isMobileDevice ? false : true
                };
            }
        }]
    });
};

beforeEach(() => {
    isMobileDevice = false;
});
describe(`Browser not supported - test`, () => {
    describe('Given desktop mode', () => {
        describe(`Given default values`, () => {
            beforeEach(() => {
                initializeWrapperDefaultValues();
            });
            it(`Should render with default desktop values`, async () => {
                await expect(renderComponent(wrapper.vm)).resolves.toMatchSnapshot();
            });
        });
        describe(`Given custom values`, () => {
            beforeEach(() => {
                initializeWrapperCustomValues();
            });

            it(`Should render with custom values provided`, async () => {
                await expect(renderComponent(wrapper.vm)).resolves.toMatchSnapshot();
            });
        });
    });

    describe(`Given mobile device size`, () => {
        beforeEach(() => {
            isMobileDevice = true;
        });
        describe(`Given default values`, () => {
            beforeEach(() => {
                initializeWrapperDefaultValues();
            });
            it(`Should render with default mobile values provided`, async () => {
                await expect(renderComponent(wrapper.vm)).resolves.toMatchSnapshot();
            });
        });
        describe(`Given custom values`, () => {
            beforeEach(() => {
                initializeWrapperCustomValues();
            });
            it(`Should render with custom mobile values provided`, async () => {
                await expect(renderComponent(wrapper.vm)).resolves.toMatchSnapshot();
            });
        });
    });
});
