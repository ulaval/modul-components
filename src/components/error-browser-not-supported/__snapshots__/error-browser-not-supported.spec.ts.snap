// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browser not supported - test Given desktop mode Given custom values Should render with custom values provided 1`] = `<div skin="warning" iconName="m-svg__warning" title="An error title." hints="aHint" links="[object Object]"></div>`;

exports[`Browser not supported - test Given desktop mode Given default values Should render with default values 1`] = `<div skin="warning" iconName="m-svg__warning" title="m-error-browser-not-supported:title" hints="m-error-browser-not-supported:hint.primary.desktop" links="[object Object]"></div>`;
